orchagent/errororch.cpp

```diff
@@ -0,0 +1,558 @@
+    return true;
+}
+
+bool ErrorOrch::deleteTableObject(string &tableName)
+{
+    SWSS_LOG_ENTER();
+
+    string appTableName = getAppTableName(tableName);
+    if(m_TableNameObjMap.find(appTableName) == m_TableNameObjMap.end())
+    {
+        SWSS_LOG_ERROR("Failed to remove EHF Table object for %s", tableName.c_str());
+        return false;
+    }
+
+    m_TableNameObjMap.erase(tableName);
+    SWSS_LOG_DEBUG("Removed EHF Table object for %s", tableName.c_str());
+
+    return true;
+}
+
+void ErrorOrch::updateErrorDb(string &tableName, const string &key,
+         std::vector<FieldValueTuple> &values)
+ {
+     SWSS_LOG_ENTER();
+     string strOp, strRc;
+     bool entryFound = false;
+     bool addEntry = false;
+     bool removeEntry = false;
+     bool updateEntry = false;
+
+     /* Extract current return code and operation */
+     for (size_t i = 0; i < values.size(); i++)
+     {
+         if(fvField(values[i]) == "operation")
+         {
+             strOp = fvValue(values[i]);
+         }
+         else if(fvField(values[i]) == "rc")
+         {
+             strRc = fvValue(values[i]);
+         }
+     }
+
+     /* Get ERROR DB table object */
+     std::shared_ptr<Table> table;
+     string errTableName = getErrorTableName(tableName);
+     if(m_TableNameObjMap.find(errTableName) == m_TableNameObjMap.end())
+     {
+         SWSS_LOG_INFO("Failed to find EHF Table object for %s", errTableName.c_str());
+         return;
+     }
+
+     table = m_TableNameObjMap[errTableName];
+     if(table == NULL)
+     {
+         SWSS_LOG_INFO("Invalid EHF Table object found for %s", errTableName.c_str());
+         return;
+     }
+
+     /* Check if the entry with the key is present in ERROR DB */
+     std::vector<FieldValueTuple> ovalues;
+     if(table->get(key, ovalues))
+     {
+         entryFound = true;
+     }
+
+     if(strRc == "SWSS_RC_SUCCESS")
+     {
+         /* Remove the entry if present in error database */
+         if(entryFound == true)
+         {
+             removeEntry = true;
+         }
+     }
+     else
+     {
+         if(strOp == "create")
+         {
+             addEntry = true;
+         }
+         else if(strOp == "remove")
+         {
+             if(entryFound == true)
+             {
+                 removeEntry = true;
+             }
+             else
+             {
+                 addEntry = true;
+             }
+         }
+         else if(strOp == "set")
+         {
+             if(entryFound == true)
+             {
+                 updateEntry = true;
+             }
+             else
+             {
+                 addEntry = true;
+             }
+         }
+     }
+     SWSS_LOG_DEBUG("key %s operation %s RC %s Entry in DB %d add:remove:update %d:%d:%d ",
+             key.c_str(), strOp.c_str(), strRc.c_str(),
+             entryFound, addEntry, removeEntry, updateEntry);
+     if(addEntry == true || updateEntry == true)
```

Comment:
If add and update are doing the same thing anyways can we just combine them into one flag to simplify the above logic?

Reply:
Joined add and update case into one.